lumicks/pylake/nb_widgets/detail/mouse.py

```diff
@@ -1,49 +1,95 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class MouseDragEvent:
+    x: float
+    y: float
+    dx: float
+    dy: float
+
+
 class MouseDragCallback:
-    def __init__(self, axes, button, callback):
-        """Callback handler to handle mouse dragging for a particular button. Callback is called with x, y, dx, dy as
-        arguments.
+    def __init__(self, axes, button, drag_callback, press_callback=None, release_callback=None):
+        """Callback handler to handle mouse dragging for a particular button. Callback is called
+        with x, y, dx, dy as arguments.
 
         Parameters
         ----------
         axes : :class:`~matplotlib.axes.Axes`
             Axes to attach the callback to.
         button : int
-            Which button to respond to (0: Left, 1: Middle, 2: Right).
-        callback : callable
-            Which function to call when dragging takes place. Note that this function should have the following input
-            signature: fun(x, y, dx, dy) where x, y are the location in data space and dx and dy are the dragging
-            distances in data space.
+            Which button to respond to (0: Left, 3: Right).
```

Comment:
For my own curiosity: Why are we now using 3 for the `Right` button?

Reply:
This is how `matplotlib` defines it.